src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -174,6 +174,8 @@ private ExportResult OnCollect(Batch<Metric> metrics)
         {
             if (this.exporter.OpenMetricsRequested)
             {
+                cursor = PrometheusSerializer.WriteTargetInfo(this.buffer, cursor, this.exporter.Resource);
```

Comment:
1) Handle the case of running out of buffer size.

2) Could we re-use the serialized data for multiple exports?

Reply:
1. Done
2. I can certainly do this, however I'm not sure if the resource attributes can change at runtime? Is this something we need to cater for? If not, we can certainly bake in the serializing of the target info. I don't know the mechanics of the resource well enough to judge.

Reply:
Resource attributes cannot be changed during runtime.